src/modules/iiif-viewer/IiifViewer.tsx

```diff
@@ -81,6 +82,28 @@ const IiifViewer = forwardRef<IiifViewerFunctions, IiifViewerProps>(
 		 * Effects
 		 */
 
+		const addFullscreenCloseButton = (openSeadragonViewer: OpenSeadragon.Viewer) => {
+			if (!openSeadragonViewer.container) {
+				return;
+			}
+			const topLeftContainer = openSeadragonViewer.container.querySelector(
+				'.openseadragon-canvas + div'
+			);
+			if (!topLeftContainer) {
+				return;
+			}
+			topLeftContainer.innerHTML = '';
+			const closeFullscreenButton = document.createElement('button');
+			closeFullscreenButton.className =
+				'p-object-detail__iiif__close-fullscreen c-button c-button--icon c-button--white';
+			closeFullscreenButton.innerHTML = 'times';
+			closeFullscreenButton.title = tText('Sluit volledig scherm');
+			closeFullscreenButton.addEventListener('click', () => {
+				openSeadragonViewer.setFullScreen(false);
+			});
+			topLeftContainer?.append(closeFullscreenButton);
+		};

```

Comment:
☝️ non-optional IMO

Reply:
that doesn't change much. All the article talks about is, not to use the renderToString function since it is too heavy.
The code below is just to render a react component to html
but i don't need a react component to be rendered to html, i just need a regular button

so i don't see the advantage of bootstrapping react to generate some button html

The reason we cannot do this fully in react is that the openseadragon is a javascript library not a react library, so it doesn't have a render function to generate custom buttons.
The other buttons we use like the fullscreen and rotate buttons exist fully in react context
but this button has to exist in the fullscreen html that the library generates when it goes fullscreen


if i needed to generate html for a react close button i would agree with you. But for an html button your solution is heavier and unnessesary

Reply:
In terms of DX and maintainability not mixing native DOM API's with frameworks that manage DOM (i.e. React) is preferable. At the very least the markup and css should be split into separate files.

In terms of performance there is no impact, React is already running.

Regardless, I don't care enough to really debate ✅